nilrt_snac/__main__.py

```diff
@@ -23,6 +22,39 @@
 """
 
 
+def _get_enabled_modules(config_file_path: Path = Path("/etc/snac/snac.conf")) -> Dict[str, bool]:
+    """Read the config file and return a dict of module enabled states. Strict validation and error reporting."""
+    enabled_modules: Dict[str, bool] = {}
+    valid_names = {c.name for c in CONFIGS}
+    if config_file_path.exists():
+        parser = configparser.ConfigParser()
+        try:
+            parser.read(str(config_file_path))
+        except (configparser.MissingSectionHeaderError, configparser.ParsingError) as e:
+            logger.error(f"Malformed config file: {config_file_path}: {e}")
+            raise SNACError(f"Malformed config file: {config_file_path}", Errors.EX_USAGE)
+        if not parser.has_section("modules"):
+            logger.warning(
+                f"Config file {config_file_path} missing [modules] section. All modules will be enabled."
+            )
+            return enabled_modules
+        for name, value in parser.items("modules"):
+            key = name.strip()
+            val = value.strip().lower()
+            if key not in valid_names:
+                logger.warning(
+                    f"Unknown module name '{key}' in config file. This entry will be ignored. Valid names: {sorted(valid_names)}"
+                )
+                continue
+            if val not in ("true", "false"):
+                logger.warning(
+                    f"Invalid value for module '{key}' in config file: '{value}'. Must be 'true' or 'false'. This module will be enabled."
```

Comment:
You say it "will be enabled" but it's not added to the list of enabled modules that this method returns. So I would expect it to read that the module "will be disabled" instead.